ingestion/src/metadata/ingestion/api/parser.py

```diff
@@ -434,3 +437,38 @@ def parse_test_connection_request_gracefully(
         )
 
     raise ParsingConfigurationError("Uncaught error when parsing the workflow!")
+
+
+def parse_ingestion_pipeline_config_gracefully(
+    config_dict: dict,
+) -> IngestionPipeline:
+    """
+    This function either correctly parses the pydantic class, or
+    throws a scoped error while fetching the required source connection
+    class.
+
+    :param config_dict: JSON ingestion pipeline config
+    :return:Ingestion Pipeline config or scoped error
+    """
+
+    try:
+        ingestion_pipeline = IngestionPipeline.parse_obj(config_dict)
+        return ingestion_pipeline
+
+    except ValidationError:
```

Comment:
should we take care of the `serviceConnection`? Not only the `sourceConfig`

Reply:
I believe that's where we get also a lot of noise, for example when we configure an URI without the scheme or something

Reply:
as discussed, that was the wrong comment. We'll be adding it to the run_automations endpoint though